Bottom: 2986de872cf3597ea2fa219c373484289a0d5f78
Top:    13df4f970ee8003bf7bfde3dede14e81508ed8bd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-03 22:02:31 +0000

Fixup frac_circle

(I think this fixes it)

Before we were getting self-intersecting polygons - a big no-no


---

diff --git a/src/polygon.c b/src/polygon.c
index e724649..bbd8e7a 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -320,7 +320,7 @@ frac_circle (PLINE * c, LocationType X, LocationType Y, Vector v, int range)
   e1 = v[0] - X;
   e2 = v[1] - Y;
 
-  range = (range == 1) ? CIRC_SEGS-1 : (CIRC_SEGS / range);
+  range = (range == 1) ? CIRC_SEGS : (CIRC_SEGS / range) - 1;
   for (i = 0; i < range; i++)
     {
       /* rotate the vector */
